packages/plugin-npm-cli/sources/commands/npm/publish.ts

```diff
@@ -155,10 +156,18 @@ async function makePublishBody(workspace: Workspace, buffer: Buffer, {access, ta
 
   const raw = await packUtils.genPackageManifest(workspace);
 
+  // This matches Lerna's logic:
+  // https://github.com/evocateur/libnpmpublish/blob/latest/publish.js#L142
+  // While the npm registry ignores the provided tarball URL, it's used by
+  // other registries such as verdaccio.
+  const tarballName = `${name}-${version}.tgz`;
+  const tarballURL = new URL(`${name}/-/${tarballName}`, registry);
+  if (tarballURL.protocol === "https:") tarballURL.protocol = "http:";
```

Comment:
🤨 What do you think of omitting this line if it works without? I don't understand it (all `tarball` fields returned from the registry are expected to be https), and it sounds counterintuitive and potentially dangerous.